modules/ballerina-core/src/main/java/org/ballerinalang/natives/typemappers/NativeCastMapper.java

```diff
@@ -134,4 +137,85 @@
+                                                       rVal.getType(), BTypes.typeInt);
+                return null; //this is unreachable practically, because above line throws an exception
+            };
+
+    public static final Function<BValueType, BValueType> ANY_TO_LONG_FUNC =
+            (rVal) -> {
+                if (rVal.getType() == BTypes.typeLong) {
+                    return rVal;
+                }
+                BLangExceptionHelper.throwRuntimeError(RuntimeErrors.CASTING_ANY_TYPE_TO_WRONG_VALUE_TYPE,
+                                                       rVal.getType(), BTypes.typeLong);
+                return null; //this is unreachable practically, because above line throws an exception
+            };
+
+    public static final Function<BValueType, BValueType> ANY_TO_FLOAT_FUNC =
+            (rVal) -> {
+                if (rVal.getType() == BTypes.typeFloat) {
+                    return rVal;
+                }
+                BLangExceptionHelper.throwRuntimeError(RuntimeErrors.CASTING_ANY_TYPE_TO_WRONG_VALUE_TYPE,
+                                                       rVal.getType(), BTypes.typeFloat);
+                return null; //this is unreachable practically, because above line throws an exception
+            };
+
+    public static final Function<BValueType, BValueType> ANY_TO_DOUBLE_FUNC =
+            (rVal) -> {
+                if (rVal.getType() == BTypes.typeDouble) {
+                    return rVal;
+                }
+                BLangExceptionHelper.throwRuntimeError(RuntimeErrors.CASTING_ANY_TYPE_TO_WRONG_VALUE_TYPE,
+                                                       rVal.getType(), BTypes.typeDouble);
+                return null; //this is unreachable practically, because above line throws an exception
+            };
+
+    public static final Function<BValueType, BValueType> ANY_TO_STRING_FUNC =
+            (rVal) -> {
+                if (rVal.getType() == BTypes.typeString) {
+                    return rVal;
+                }
+                BLangExceptionHelper.throwRuntimeError(RuntimeErrors.CASTING_ANY_TYPE_TO_WRONG_VALUE_TYPE,
+                                                       rVal.getType(), BTypes.typeString);
+                return null; //this is unreachable practically, because above line throws an exception
+            };
+
+    public static final Function<BValueType, BValueType> ANY_TO_BOOLEAN_FUNC =
+            (rVal) -> {
+                if (rVal.getType() == BTypes.typeBoolean) {
+                    return rVal;
+                }
+                BLangExceptionHelper.throwRuntimeError(RuntimeErrors.CASTING_ANY_TYPE_TO_WRONG_VALUE_TYPE,
+                                                       rVal.getType(), BTypes.typeBoolean);
+                return null; //this is unreachable practically, because above line throws an exception
+            };
+
+    public static final Function<BValueType, BValueType> ANY_TO_JSON_FUNC =
```

Comment:
ANY TO JSON and ANY TO XML will fail, because those values does not inherit from BValueType